js/gate.js

```diff
@@ -1824,13 +1828,18 @@ module.exports = class gate extends Exchange {
         //         'with_id': true, // return order book ID
         //     };
         //
-        const [ request, query ] = this.prepareRequest (market, undefined, params);
-        const method = this.getSupportedMapping (market['type'], {
+        let request = undefined;
+        [ request, params ] = this.prepareRequest (market, undefined, params);
+        let method = this.getSupportedMapping (market['type'], {
```

Comment:
@Dan-krm methods like this or fetchTicker/etc where we have a symbol and the spot and spot-margin endpoint are the same, we don't need to do these extra checks, what's the difference between this call

`fetchTicker("BTC/USDT", params={"marginMode":"isolated"})`

and 

`fetchTicker("BTC/USDT")`

or 

`fetchTicker("BTC/USDT", params={"marginMode":"cross"})`

there is no difference the same endpoint is used, so I'd suggest removing the margin handling from them, let me know what you think

Reply:
That makes sense, I removed handleMarginModeAndParams from methods that require a symbol and use the same endpoint and request parameters for spot and spot-margin